test/test_torch.py

```diff
@@ -4178,6 +4178,36 @@ def logcumsumexp(a, axis):
                 'expected scalar_type Double but found Float'):
             torch.logcumsumexp(b, axis, out=inplace_out)
 
+    def test_diff(self, device):
```

Comment:
Add a test for bool, too, and another test comparing directly with NumPy's np.diff for bool, long, float, cfloat.